documentation/USAGE.rst

```diff
@@ -535,68 +535,223 @@ Further demonstrations of spatially varying boundary condition can be found
 completely dominate the matrix diagonal and the RHS vector in cells
 that are masked. The ``mask`` variable would typically be a
 ``CellVariable`` boolean constructed using the cell center values.
 
-One must be careful to distinguish between constraining internal cell
-values during the solve step and simply applying arbitrary constraints
-to a ``CellVariable``. Applying a constraint,
+Internal fixed gradient
+-----------------------
+
+An equation of the form
+
+>>> eqn = TransientTerm() == DiffusionTerm(coeff=Gamma0)
+
+can be constrained to have a fixed internal ``gradient`` magnitude 
+at a position given by ``mask`` with the following alterations
+
+>>> Gamma = FaceVariable(mesh=mesh, value=Gamma0)
+>>> Gamma[mask.value] = 0.
+>>> eqn = (TransientTerm() == DiffusionTerm(coeff=Gamma) 
+...        + DiffusionTerm(coeff=largeValue * mask)
+...        - ImplicitSourceTerm(mask * largeValue * gradient 
+...                             * mesh.faceNormals).divergence)
+
+The parameter ``largeValue`` must be chosen to be large enough to
+completely dominate the matrix diagonal and the RHS vector in cells
+that are masked. The ``mask`` variable would typically be a
+``FaceVariable`` boolean constructed using the face center values.
+
+Internal Robin condition
+------------------------
+
+Nothing different needs to be done when 
+`applying Robin boundary conditions`_ at internal faces.
+
+.. note::
+
+   While we believe these derivations are "correct", they often do not 
+   seem to produce the intuitive result. At this point, we think this has 
+   to do with the pathology of "internal" boundary conditions, but remain 
+   open to other explanations. :term:`FiPy` was designed with diffuse 
+   interface treatments (phase field and level set) in mind and, as such, 
+   internal "boundaries" do not come up in our own work and have not 
+   received much attention.
+
+.. note::
+
+  One must be careful to distinguish between constraining internal cell
+  values during the solve step and simply applying arbitrary constraints
+  to a ``CellVariable``. Applying a constraint,
 
->>> var.constrain(value, where=mask)
+  >>> var.constrain(value, where=mask)
 
-simply fixes the returned value of ``var`` at ``mask`` to be
-``value``. It does not have any effect on the implicit value of ``var`` at the
-``mask`` location during the linear solve so it is not a substitute
-for the source term machinations described above. Future releases of
-:term:`FiPy` may implicitly deal with this discrepancy, but the current
-release does not. A simple example can be used to demonstrate this::
+  simply fixes the returned value of ``var`` at ``mask`` to be
+  ``value``. It does not have any effect on the implicit value of ``var`` at the
+  ``mask`` location during the linear solve so it is not a substitute
+  for the source term machinations described above. Future releases of
+  :term:`FiPy` may implicitly deal with this discrepancy, but the current
```

Comment:
If there is not an issue for fixing this, please open one, or remove the time-dependence of the explanation.

Reply:
Filed #581